homeassistant/components/romy/vacuum.py

```diff
@@ -0,0 +1,146 @@
+https://home-assistant.io/components/vacuum.romy/.
+"""
+
+
+from typing import Any
+
+from romy import RomyRobot
+
+from homeassistant.components.vacuum import StateVacuumEntity, VacuumEntityFeature
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+
+from .const import DOMAIN, ICON, LOGGER
+from .coordinator import RomyVacuumCoordinator
+
+FAN_SPEED_NONE = "default"
+FAN_SPEED_NORMAL = "normal"
+FAN_SPEED_SILENT = "silent"
+FAN_SPEED_INTENSIVE = "intensive"
+FAN_SPEED_SUPER_SILENT = "super_silent"
+FAN_SPEED_HIGH = "high"
+FAN_SPEED_AUTO = "auto"
+
+FAN_SPEEDS: list[str] = [
+    FAN_SPEED_NONE,
+    FAN_SPEED_NORMAL,
+    FAN_SPEED_SILENT,
+    FAN_SPEED_INTENSIVE,
+    FAN_SPEED_SUPER_SILENT,
+    FAN_SPEED_HIGH,
+    FAN_SPEED_AUTO,
+]
+
+# Commonly supported features
+SUPPORT_ROMY_ROBOT = (
+    VacuumEntityFeature.BATTERY
+    | VacuumEntityFeature.PAUSE
+    | VacuumEntityFeature.RETURN_HOME
+    | VacuumEntityFeature.STATE
+    | VacuumEntityFeature.START
+    | VacuumEntityFeature.STOP
+    | VacuumEntityFeature.FAN_SPEED
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config_entry: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up ROMY vacuum cleaner."""
+
+    coordinator: RomyVacuumCoordinator = hass.data[DOMAIN][config_entry.entry_id]
+    romy: RomyRobot = coordinator.romy
+
+    device_info = {
+        "manufacturer": "ROMY",
+        "model": romy.model,
+        "sw_version": romy.firmware,
+        "identifiers": {"serial": romy.unique_id},
+    }
+
+    romy_vacuum_entity = RomyVacuumEntity(coordinator, romy, device_info)
+    entities = [romy_vacuum_entity]
+    async_add_entities(entities, True)
+
+
+class RomyVacuumEntity(CoordinatorEntity[RomyVacuumCoordinator], StateVacuumEntity):
+    """Representation of a ROMY vacuum cleaner robot."""
+
+    _attr_translation_key = DOMAIN
+    _attr_has_entity_name = True
+
+    def __init__(
+        self,
+        coordinator: RomyVacuumCoordinator,
+        romy: RomyRobot,
+        device_info: dict[str, Any],
+    ) -> None:
+        """Initialize the ROMY Robot."""
+        super().__init__(coordinator)
+        self.romy = romy
+        self._device_info = device_info
+        self._attr_unique_id = self.romy.unique_id
+        self._attr_supported_features = SUPPORT_ROMY_ROBOT
+        self._attr_fan_speed_list = FAN_SPEEDS
+        self._attr_icon = ICON
+        self._is_on = False
+
+    @callback
+    def _handle_coordinator_update(self) -> None:
+        """Handle updated data from the coordinator."""
+        self._attr_fan_speed = FAN_SPEEDS[self.romy.fan_speed]
+        self._attr_battery_level = self.romy.battery_level
+        self._attr_state = self.romy.status
+        self._attr_name = self.romy.name
```

Comment:
```suggestion
```

If the name of the device really can change, you should update the device info

Reply:
update now device info too, but let the name update too,
in case someone has two romy robots its better so to differentiate them

Reply:
Please implement the suggestion as the `_attr_has_entity_name` prefixes each entity by the device name.

Without removing the name here, the entity would get name `Susi Susi`, when you name the bot `Susi`

Reply:
I don't see double names when i test it

current status (i see no double names here):
![image](https://github.com/home-assistant/core/assets/2132412/e2132409-6ffb-4880-acb5-49d6bc970e8e)

![image](https://github.com/home-assistant/core/assets/2132412/38f5daf8-9c22-4f7c-a7c3-966724d8ab58)


without the line (self._attr_name = self.romy.name)
![image](https://github.com/home-assistant/core/assets/2132412/a7a6c2dc-6d10-40f3-98f1-24b754bd6bdd)

Reply:
changed it, unique_id is now shown as name, can i somehow easy adjust it and show robot name instead?